Lib/test/test_bytes.py

```diff
@@ -1359,6 +1359,38 @@ def by(s):
         b = by("Hello, world")
         self.assertEqual(re.findall(br"\w+", b), [by("Hello"), by("world")])
 
+    def test_resize(self):
+        ba = bytearray(b'abcdef')
+        self.assertIsNone(ba.resize(3))
+        self.assertEqual(ba, bytearray(b'abc'))
+        self.assertIsNone(ba.resize(10))
+        self.assertEqual(len(ba), 10)
+        # Bytes beyond set values must be cleared.
+        self.assertEqual(ba, bytearray(b'abc\0\0\0\0\0\0\0'))
+        ba[3:10] = b'defghij'
+        self.assertEqual(ba, bytearray(b'abcdefghij'))
+        self.assertIsNone(ba.resize(2**20))
+        self.assertEqual(len(ba), 2**20)
+        self.assertEqual(ba, bytearray(b'abcdefghij' + b'\0' * (2 ** 20 - 10)))
+        self.assertIsNone(ba.resize(0))
+        self.assertEqual(ba, bytearray())
+        self.assertIsNone(ba.resize(10))
+        self.assertEqual(ba, bytearray(b'\0' * 10))
+
+        ba = ByteArraySubclass(b'abcdef')
+        self.assertIsNone(ba.resize(3))
+        self.assertEqual(ba, bytearray(b'abc'))
+
+        # Check arguments
+        self.assertRaises(TypeError, lambda: bytearray().resize())
+        self.assertRaises(TypeError, bytearray().resize, 10, 10)
+
+        self.assertRaises(BufferError, lambda: bytearray().resize(-1))
```

Comment:
What is a `BufferError`?  It looks like the C code calls `PyErr_Format(PyExc_ValueError` which is what I'd expect.

Reply:
Yep, already changing / Victor suggested as well :).